src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,267 @@
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+
+public class Duke {
+
+    private static ArrayList<Task> taskList = new ArrayList<>();
+    public static final String PATHNAME = "data/duke.txt";
+
+    private static void writeToFile(String path) throws IOException {
+        FileWriter fw = new FileWriter(path);
+        for (int i = 0; i < taskList.size(); i++) {
+            fw.write(taskList.get(i).printToFile());
+        }
+        fw.close();
+    }
+
+    public static void addTask(Task task, boolean isPrint) {
+        taskList.add(task);
+        try {
+            writeToFile(PATHNAME);
+        } catch (IOException ioException) {
+            System.out.println("Error updating file");
+        }
+        if (isPrint) {
+            System.out.printf("Got it. I've added this task:\n%s\nNow you have %d tasks in the list.\n",
+                    task.print(), taskList.size());
+        }
+    }
+
+    private static void loadFromFile(String path) throws FileNotFoundException {
+        File f = new File(path);
+        System.out.println("Path: " + f.getAbsolutePath());
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String[] words = s.nextLine().split("\\|");
+            switch (words[0]) {
+            case "T":
+                Todo todo = new Todo(words[2], words[1].equals("1"));
+                addTask(todo, false);
+                break;
+            case "D":
+                Deadline deadline = new Deadline(words[2], words[1].equals("1"), words[3]);
+                addTask(deadline, false);
+                break;
+            case "E":
+                Event event = new Event(words[2], words[1].equals("1"), words[3]);
+                addTask(event, false);
+                break;
+            default:
+                System.out.println("Error reading data from file: Invalid format");
+                break;
+            }
+        }
+    }
+
+    public  static  void todo(String line, String[] words) {
+        if (words.length >= 2) {
+            String task = line.substring(line.indexOf(' ') + 1);
+            Todo todo = new Todo(task, false);
+            addTask(todo, true);
+        } else {
+            System.out.println("Usage: todo <task>");
+        }
+    }
+
+    public static void deadline(String line) {
+        try {
+            String task = line.substring(line.indexOf(' ')+1, line.indexOf("/by")-1);
+            String date = line.substring(line.indexOf("/by")+4);
+            Deadline deadline = new Deadline(task, false, date);
+            addTask(deadline, true);
+        } catch (StringIndexOutOfBoundsException stringIndexOutOfBoundsException) {
+            System.out.println("Usage: deadline <task> /by <date>");
+        }
+    }
+
+    public static void event(String line) {
+        try {
+            String task = line.substring(line.indexOf(' ')+1, line.indexOf("/at")-1);
+            String date = line.substring(line.indexOf("/at")+4);
```

Comment:
Likewise, you can also consider relocating the `'/at'` keyword to a variable too!